Task2/circularList.c

```diff
@@ -0,0 +1,126 @@
+#include "circularList.h"
+
+struct Node
+{
+    int value;
+    struct Node* nextNodePtr;
+};
+
+struct List
+{
+    Node* headPtr;
+    Node* tailPtr;
+};
+
+int createList(List** ptrToListPtr)
+{
+    *ptrToListPtr = calloc(1, sizeof(List));
+    if (*ptrToListPtr == NULL)
+    {
+        return -1;
+    }
+    (*ptrToListPtr)->headPtr = NULL;
+    (*ptrToListPtr)->tailPtr = NULL;
+    return 0;
+}
+
+bool isListEmpty(List* listPtr)
+{
+    return listPtr->headPtr == NULL;
+}
+
+bool isSingleNodeInList(List* listPtr)
+{
+    return (listPtr->headPtr == listPtr->tailPtr);
```

Comment:
```suggestion
    return listPtr->headPtr == listPtr->tailPtr;
```
А если `listPtr` является NULL?
И необязательные скобки которые лучше опустить